sys-account/server/authn/pkg/auth/token.go

```diff
@@ -0,0 +1,118 @@
+package auth
+
+import (
+	"github.com/getcouragenow/packages/sys-account/authn/pkg/utilities"
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/getcouragenow/packages/sys-account/rpc"
+)
+
+var (
+	JwtAccessSecret   = []byte(os.Getenv("JWT_ACCESS_SECRET"))
+	JwtRefreshSecret  = []byte(os.Getenv("JWT_REFRESH_SECRET"))
+	AccessExpiration  = 10 * time.Minute    // 10 minutes access token
+	RefreshExpiration = 14 * 24 * time.Hour // two weeks
+)
+
+// Claimants are ones who are able to get token claims
+type Claimant interface {
+	GetEmail() string
+	GetRole() *rpc.UserRoles
+}
+
+// TokenClaims is the representation of JWT auth claims
+type TokenClaims struct {
+	UserId    string         `json:"userId"`
+	Role      *rpc.UserRoles `json:"role"`
+	UserEmail string         `json:"userEmail"`
+	jwt.StandardClaims
+}
+
+// TokenPairDetails contain both AccessToken and RefreshToken of the user
+// This in turn will be saved to the Genji / Badger DB as `session data`
+type TokenPairDetails struct {
```

Comment:
Refresh and Access token is going to be inserted to the Genji / Badger for managing sessions, i'm all ears if anyone has second opinion on this.

Reply:
Not a JWT expert but sounds like the right place.

I plan to store user images in the DB also. Why ? so that stuff is not spread all over the Disk and so when we add RAFT to genj later all servers have the same data.

The Lets Encrypt will cert that is generated will also be stored in Genji.

So in summary what i am saying is that any state should be in genji, so storing the Refresh and AccessToken in the DB looks like the correct approach that is in synergy with our architectural and Ops goals.